### PR TITLE
Query store for supported series of charm

### PR DESCRIPTION
**Pull request by wallyworld**
Add support for querying the supported-series for a charm when resolving. The requires the new charm store.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/132/
Test PASSed.

File: charmstore_test.go [hunk 1]
```diff
@@ -506,33 +506,41 @@ func (s *charmStoreRepoSuite) TestResolve(c *gc.C) {
 
 	// Define the tests to be run.
 	tests := []struct {
```
Comment (rogpeppe) [inline charmstore_test.go]: Can we have at least one test that tests the results for a multi-series charm, please?

**Comment by rogpeppe** [comment]
LGTM with another test or two. Thanks!

File: charmstore_test.go (hunk 1 above)
Comment (wallyworld) [inline charmstore_test.go]: Done

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/134/
Test PASSed.

**Comment by wallyworld** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Pull request merged**